packages/blocks/src/page-block/edgeless/components.ts

```diff
@@ -188,3 +166,159 @@ export function EdgelessBlockChildrenContainer(
+      ];
+      const rightCenter = [
+        rect.x + rect.width + PADDING_X * this.zoom,
+        rect.y + rect.height / 2 + (PADDING_Y * this.zoom) / 2,
+      ];
+      const handleLeft = Handle(
+        leftCenter[0],
+        leftCenter[1],
+        'left',
+        this._onHandleMouseDown
+      );
+      const handleRight = Handle(
+        rightCenter[0],
+        rightCenter[1],
+        'right',
+        this._onHandleMouseDown
+      );
+      handles = html` ${handleLeft}${handleRight} `;
+    }
+    return handles;
+  }
+
+  private _onHandleMouseDown = (e: MouseEvent, type: 'left' | 'right') => {
+    // prevent selection action being fired
+    e.stopPropagation();
+    if (this.state?.type !== 'none') {
+      const {
+        rect,
+        selected: { xywh },
+      } = this.state;
+      const [x] = JSON.parse(xywh) as XYWH;
+      this._dragStartInfo = {
+        startMouseLeft: e.clientX,
+        absoluteLeft: x,
+        // the width of the selected group may 0 after init use rect.width instead
+        width: rect.width,
+        direction: type,
+      };
+      // parent ele is the edgeless block container
+      this.parentElement?.addEventListener('mousemove', this._onDragMove);
+      this.parentElement?.addEventListener('mouseup', this._onDragEnd);
+    }
+  };
+
+  private _onDragMove = (e: MouseEvent) => {
+    let newX = 0;
+    let newW = 0;
+    if (this.state.type === 'single') {
+      const {
+        selected,
+        viewpoint
+      } = this.state;
+      const { xywh } = selected
+      const [x, y, w] = JSON.parse(xywh) as XYWH;
+      const minus = this._dragStartInfo.startMouseLeft - e.clientX;
+      if (this._dragStartInfo.direction === 'left') {
+        newX = this._dragStartInfo.absoluteLeft - minus / this.zoom;
+        newW = (this._dragStartInfo.width + minus) / this.zoom;
+      } else {
+        newX = x;
+        newW = (this._dragStartInfo.width - minus) / this.zoom;
+      }
+      // limit the width of the selected group
+      if (newW < GROUP_MIN_LENGTH) {
+        newW = GROUP_MIN_LENGTH;
+        newX = x;
+      }
+      // if xywh do not change, no need to update
+      if (newW === w && newX === x) {
+        return;
+      }
+      const groupBlock = getBlockById<'div'>(selected.id);
+      const groupContainer = groupBlock?.parentElement;
+      // first change container`s x/w directly for get groups real height
+      if (groupContainer) {
+        groupContainer.style.width = newW + 'px';
+        groupContainer.style.translate = `translate(${newX}px, ${y}px) scale(${this.zoom})`;
+      }
+      // reset the width of the container may trigger animation
+      requestAnimationFrame(() => {
+        // refresh xywh by model
+        if (!this.lock) {
+          selected.page.captureSync();
+          this.lock = true;
+        }
+        if (this.state.type !== 'none') {
+          this.state.rect = getSelectionBoxBound(viewpoint, selected.xywh);
```

Comment:
The reason is I have to sync the position of the selection rect. I didn't find any way to access the `this._document.viewpoint`

Reply:
Just curious to know if `viewpoint` is a typo of `viewport` here?

Reply:
😨

Reply:
I have only the primary school degree(